dashboard/modules/reporter/reporter_agent.py

```diff
@@ -369,6 +372,25 @@ async def CpuProfiling(self, request, context):
         )
         return reporter_pb2.CpuProfilingReply(output=output, success=success)
 
+    async def MemoryProfiling(self, request, context):
+        pid = request.pid
+        format = request.format
+        leaks = request.leaks
+        duration = request.duration
+        native = request.native
+        p = MemoryProfilingManager(self._log_dir)
+        success, output = await p.attach_profiler(pid, native=native)
+        profiler_filename = output.split("\n")[-1]
+        if not success:
+            return reporter_pb2.MemoryProfilingReply(output=output, success=success)
+
+        await asyncio.sleep(duration + 1)  # add 1 second overhead
+        success, output = await p.detach_profiler(pid)
```

Comment:
We should handle the case detach fail -> get profile result success. Only detach success -> get profile fail is covered now. Also need tests for this case

Reply:
detach fail and get profile result success should be covered too since we ignore whether success is true or false here. In get profile result, we check again if the profiled file exists or not.